src/main/java/com/gregtechceu/gtceu/GTCEu.java

```diff
@@ -53,45 +61,132 @@ public static ResourceLocation appendId(String id) {
         return new ResourceLocation(strings[0], strings[1]);
     }
 
-    public static boolean isKubeJSLoaded() {
-        return LDLib.isModLoaded(GTValues.MODID_KUBEJS);
+    /**
+     * @return if we're running in a production environment
+     */
+    public static boolean isProd() {
+        return FMLLoader.isProduction();
     }
 
-    public static boolean isIrisOculusLoaded() {
-        return LDLib.isModLoaded(GTValues.MODID_IRIS) || LDLib.isModLoaded(GTValues.MODID_OCULUS);
+    /**
+     * @return if we're not running in a production environment
+     */
+    public static boolean isDev() {
+        return !isProd();
     }
 
-    public static boolean isSodiumRubidiumEmbeddiumLoaded() {
-        return LDLib.isModLoaded(GTValues.MODID_SODIUM) || LDLib.isModLoaded(GTValues.MODID_RUBIDIUM) ||
-                LDLib.isModLoaded(GTValues.MODID_EMBEDDIUM);
+    /**
+     * @return if we're running data generation
+     */
+    public static boolean isDataGen() {
+        return FMLLoader.getLaunchHandler().isData();
     }
 
-    public static boolean isAE2Loaded() {
-        return LDLib.isModLoaded(GTValues.MODID_APPENG);
+    /**
+     * A friendly reminder that the server instance is populated on the server side only, so null/side check it!
+     * 
+     * @return the current minecraft server instance
+     */
+    public static MinecraftServer getMinecraftServer() {
+        return ServerLifecycleHooks.getCurrentServer();
     }
 
-    public static boolean isCuriosLoaded() {
-        return LDLib.isModLoaded(GTValues.MODID_CURIOS);
+    /**
+     * @param modId the mod id to check for
+     * @return if the mod whose id is {@code modId} is loaded or not
+     */
+    public static boolean isModLoaded(String modId) {
+        return ModList.get().isLoaded(modId);
     }
 
-    public static boolean isShimmerLoaded() {
-        return LDLib.isModLoaded(GTValues.MODID_SHIMMER);
+    /**
+     * For async stuff use this, otherwise use {@link GTCEu isClientSide}
+     * 
+     * @return if the current thread is the client thread
+     */
+    public static boolean isClientThread() {
+        return isClientSide() && Minecraft.getInstance().isSameThread();
     }
 
-    public static boolean isJAVDLoaded() {
-        return LDLib.isModLoaded(GTValues.MODID_JAVD);
+    /**
+     * @return if the FML environment is a client
+     */
+    public static boolean isClientSide() {
+        return FMLEnvironment.dist.isClient();
     }
 
-    public static boolean isFTBTeamsLoaded() {
-        return LDLib.isModLoaded(GTValues.MODID_FTB_TEAMS);
+    /**
+     * This check isn't the same for client and server!
+     * 
+     * @return if it's safe to access the current instance {@link net.minecraft.world.level.Level Level} on client or if
+     *         it's safe to access any level on server.
+     */
+    public static boolean canGetServerLevel() {
```

Comment:
why is this named `canGetServerLevel` but does client checks too?
also, is this safe to run on a dedicated server (because of aforementioned clientside class usage)?

Reply:
it should be, we used to use `Platform.isServerNotSafe()` which is just the negation of this method.
Although I should try to run it on a server and see if anything breaks.

Reply:
Should maybe rename it to `isLevelSafe()` or something to make it clearer